packaging_automation/common_tool_methods.py

```diff
@@ -0,0 +1,286 @@
+                return True
+    return False
+
+
+def count_line_in_file(base_path: str, relative_file_path: str, search_line: str) -> int:
+    with open(f"{base_path}/{relative_file_path}", "r") as reader:
+        content = reader.read()
+        lines = content.splitlines()
+    return len(list(filter(lambda line: line == search_line, lines)))
+
+
+def replace_line_in_file(file: str, match_regex: str, replace_str: str) -> bool:
+    with open(file, "r") as reader:
+        file_content = reader.read()
+        lines = file_content.splitlines()
+        has_match = False
+        for line_number, line in enumerate(lines):
+            if re.match(match_regex, line.strip()):
+                has_match = True
+                lines[line_number] = replace_str
+        edited_content = str_array_to_str(lines)
+    with open(file, "w") as writer:
+        writer.write(edited_content)
+
+    return has_match
+
+
+def append_line_in_file(file: str, match_regex: str, append_str: str) -> bool:
+    with open(file, "r+") as reader:
+        file_content = reader.read()
+        lines = file_content.splitlines()
+        has_match = False
+        copy_lines = lines.copy()
+        appended_line_index = 0
+        for line_number, line in enumerate(lines):
+            if re.match(match_regex, line.strip()):
+                has_match = True
+
+                if line_number + 1 < len(lines):
+                    copy_lines[appended_line_index + 1] = append_str
+                    lines_to_be_shifted = lines[line_number + 1:]
+                    copy_lines = copy_lines[0:appended_line_index + 2] + lines_to_be_shifted
+                else:
+                    copy_lines.append(append_str)
+            appended_line_index = appended_line_index + 1
+        edited_content = str_array_to_str(copy_lines)
+    with open(file, "w") as writer:
+        writer.write(edited_content)
+
+    return has_match
+
+
+def prepend_line_in_file(file: str, match_regex: str, append_str: str) -> bool:
+    with open(file, "r+") as reader:
+        file_content = reader.read()
+        lines = file_content.splitlines()
+        has_match = False
+        copy_lines = lines.copy()
+        prepended_line_index = 0
+        for line_number, line in enumerate(lines):
+            if re.match(match_regex, line.strip()):
+                has_match = True
+                copy_lines[prepended_line_index] = append_str
+                lines_to_be_shifted = lines[line_number:]
+                copy_lines = copy_lines[0:prepended_line_index + 1] + lines_to_be_shifted
+            prepended_line_index = prepended_line_index + 1
+        edited_content = str_array_to_str(copy_lines)
+    with open(file, "w") as writer:
+        writer.write(edited_content)
+
+    return has_match
+
+
+def get_current_branch(working_dir: str) -> str:
+    repo = get_new_repo(working_dir)
+    return repo.active_branch
+
+
+def does_remote_branch_exist(branch_name: str, working_dir: str) -> bool:
```

Comment:
minor:

`does_` prefix sounds a bit unusual to me, maybe:

does_remote_branch_exist -> remote_branch_exists
does_local_branch_exist -> local_branch_exists
does_branch_exist -> branch_exists

Reply:
Fixed